src/app/core/services/remote-config.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable } from '@angular/core'
+import { Firebase } from '@ionic-native/firebase/ngx'
+
+import { BehaviorSubject, from, Observable } from 'rxjs'
+import { ConfigKeys } from '../../shared/enums/config'
+import 'rxjs/add/operator/mergeMap'
+import { LogService } from './log.service'
+
+@Injectable()
+export class RemoteConfigService {
+  setCacheExpiration(timeoutMillis: number) {
+    throw new Error("RemoteConfigService method not implemented")
+  }
+
+  read(): Promise<RemoteConfig> {
+    throw new Error("RemoteConfigService method not implemented")
+  }
+
+  forceFetch(): Promise<RemoteConfig> {
+    throw new Error("RemoteConfigService method not implemented")
+  }
+
+  subject(): Observable<RemoteConfig> {
+    throw new Error("RemoteConfigService method not implemented")
+  }
+}
+
+export interface RemoteConfig {
+  get(key: ConfigKeys): Promise<any>
+  getOrDefault(key: ConfigKeys, defaultValue: any): Promise<any>
+}
+
+class FirebaseRemoteConfig implements RemoteConfig {
+  constructor(
+    private firebase: Firebase,
+    private logger: LogService,
+  ) {}
+
+  get(key: ConfigKeys): Promise<string | null> {
+    return this.firebase.getValue(key.value, '')
+  }
+
+  getOrDefault(key: ConfigKeys, defaultValue: string): Promise<string> {
+    console.log(`Retrieving ${key.value}`)
+    return this.firebase.getValue(key.value, '')
+      .then((val: string) => val.length == 0 ? defaultValue : val)
+      .catch(e => {
+        this.logger.error(`Failed to retrieve ${key.value} (using default ${defaultValue})`, e)
+        return defaultValue
+      })
+  }
+}
+
+@Injectable()
+export class FirebaseRemoteConfigService extends RemoteConfigService {
+  private timeoutMillis = 14_400_000;  // 3 hours
+  private lastFetch: number = 0;
```

Comment:
Should this maybe be stored in the storage instead since it resets every time the app is closed?

Reply:
Done!

Reply:
What about `lastFetch`? Maybe this could be stored as well or did you want it to refresh every time the app is opened?

Reply:
This can be used if we move to our own config server, however, Firebase config is not necessarily directly accessible across app boots, so after a new app gets started it's fine to fetch it again. In any case, Firebase itself has an additional layer of caching behind this. This plugin unfortunately does not allow fetching all remote config because it does not expose a way to fetch all configured keys, otherwise we could persist the entire config in storage. Again, this is something we COULD do with radar-app-server.

Reply:
Oh okay